core/src/test/java/org/elasticsearch/index/get/GetResultTests.java

```diff
@@ -0,0 +1,188 @@
+                expectedFields = tuple.v2();
+            }
+        } else {
+            version = -1;
+            exists = false;
+        }
+        GetResult getResult = new GetResult(index, type, id, version, exists, source, fields);
+        GetResult expectedGetResult = new GetResult(index, type, id, version, exists, source, expectedFields);
+        return Tuple.tuple(getResult, expectedGetResult);
+    }
+
+    private static Tuple<Map<String, GetField>,Map<String, GetField>> randomGetFields(XContentType xContentType) {
+        int numFields = randomIntBetween(2, 10);
+        Map<String, GetField> fields = new HashMap<>(numFields);
+        Map<String, GetField> expectedFields = new HashMap<>(numFields);
+        for (int i = 0; i < numFields; i++) {
+            Tuple<GetField, GetField> tuple = randomGetField(xContentType);
+            GetField getField = tuple.v1();
+            GetField expectedGetField = tuple.v2();
+            fields.put(getField.getName(), getField);
+            expectedFields.put(expectedGetField.getName(), expectedGetField);
+        }
+        return Tuple.tuple(fields, expectedFields);
+    }
+
+    //TODO move this to some utility class, this should be useful whenever we have to test parsing _source
+    public static BytesReference randomSource() {
+        try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
+            builder.startObject();
+            addFields(builder, 0);
+            builder.endObject();
+            return builder.bytes();
+        } catch(IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void addFields(XContentBuilder builder, int currentDepth) throws IOException {
+        int numFields = randomIntBetween(1, 5);
+        for (int i = 0; i < numFields; i++) {
+            if (frequently()) {
+                builder.field(randomAsciiOfLengthBetween(3, 10), randomFieldValue());
+            }
+            if (randomBoolean() && currentDepth < 5) {
+                builder.startObject(randomAsciiOfLengthBetween(3, 10));
+                addFields(builder, currentDepth + 1);
+                builder.endObject();
+            }
+            if (randomBoolean() && currentDepth < 5) {
+                builder.startArray(randomAsciiOfLengthBetween(3, 10));
+                int numElements = randomIntBetween(1, 5);
+                boolean object = randomBoolean();
+                for (int j = 0; j < numElements; j++) {
+                    if (object) {
+                        builder.startObject();
+                        addFields(builder, currentDepth + 1);
+                        builder.endObject();
+                    } else {
+                        builder.value(randomFieldValue());
+                    }
+                }
+                builder.endArray();
+            }
+        }
+    }
+
+    private static Object randomFieldValue() {
+        switch(randomIntBetween(0, 3)) {
```

Comment:
it's more concise, but we end up generating many random values that we don't use. We keep only one. I think I prefer the old fashioned switch.